boot/bootstate20.go

```diff
@@ -546,7 +596,7 @@ func (ks20 *bootState20Kernel) commit() error {
 // modeenv, but no state needs to be committed when choosing to mount a
 // kernel snap.
 func (ks20 *bootState20Kernel) selectAndCommitSnapInitramfsMount() (sn snap.PlaceInfo, err error) {
-	err = ks20.kModeenv.loadModeenv()
+	err = ks20.loadModeenv()
```

Comment:
We will need to think a bit about this ones and removing the Modeenv from inside the bootState20Base|Kernel. Maybe this can use Update as well, but we need to make sure we turn off sealing. Anyway a "purpose" flag on *Update could help add some usage checks to it.

Reply:
I think for now I will leave out this purpose flag and we can sync up on that a bit after this has landed because it's a little unclear to me how this flow would work, because really we can get called into this in a few different starting places so it's difficult to reuse the modeenv if it is passed in from outside without making `if uc20 { getModeenv() }` inside the boot.MarkSuccessful properly, etc. which is kinda awkward

Reply:
yes, that's fine, but I mentioned this because it relates to moving the Modeenv out of the *States, and we'll need to think about that